### ISSUE TITLE
`expect.any` mutates object

### ISSUE THREAD

**Issue by Electroid** [oven-sh/bun#8109]
```js
test("can pass null as option to stdio", () => {
  const result = spawnSync(bunExe(), { stdio: [null, null, null] });
  console.log(result);
  expect(result).toMatchObject({
    status: 0,
    signal: null,
    output: [null, expect.any(Buffer), expect.any(Buffer)],
    stdout: expect.any(Buffer),
    stderr: expect.any(Buffer),
  });
  console.log(result);
});
```

First log:
```js
{
  signal: null,
  status: 0,
  output: [ null, Buffer(1881) [ ... 1368 more ],
    Buffer(0) [  ]
  ],
  stdout: Buffer(1881) [ ... 1368 more ],
  stderr: Buffer(0) [  ],
}
```

Second log:
```js
{
  signal: null,
  status: 0,
  output: [ null, Any<Function>, Any<Function> ],
  stdout: Any<Function>,
  stderr: Any<Function>,
}
```

**Comment by keverw** [comment]
I just bumped into this too. I have a class that does something, and then if it was already done won't do it again. So I check the same instance shared between the test cases. I also came about #3521 when I found this bug, and I assume that one is probably the same underlying issue.

For anyone else running into this issue: My workaround was just to use `cloneDeep` from Lodash as a workaround before passing it into expect. But there are other ways to clone too.

```
expect(cloneDeep(prettierHelpers.config)).toMatchSnapshot({
  filePath: expect.any(String),
});
```
